packages/paste-libraries/react-textarea-autosize/src/__tests__/index.test.tsx

```diff
@@ -0,0 +1,133 @@
+    expect(element.style.position).toEqual('absolute');
+    expect(element.style.zIndex).toEqual('-1000');
+    expect(element.style.top).toEqual('0px');
+    expect(element.style.right).toEqual('0px');
+  });
+});
+
+describe('getSizingData', () => {
+  it('returns the sizing styles on an HTML node', () => {
+    const element = document.createElement('textarea');
+    // need to set the box sizing property because the function returns null if it is not set
+    element.style.setProperty('box-sizing', 'border-box');
+    const sizingData = getSizingData(element);
+
+    expect(sizingData).toEqual({
+      borderSize: 2,
+      paddingSize: 4,
+      sizingStyle: {
+        borderBottomWidth: '1px',
+        borderLeftWidth: '1px',
+        borderRightWidth: '1px',
+        borderTopWidth: '1px',
+        boxSizing: 'border-box',
+        fontFamily: '-webkit-small-control',
+        fontSize: '',
+        fontStyle: '',
+        fontWeight: '',
+        letterSpacing: 'normal',
+        lineHeight: 'normal',
+        paddingBottom: '2px',
+        paddingLeft: '2px',
+        paddingRight: '2px',
+        paddingTop: '2px',
+        tabSize: '',
+        textIndent: '0',
+        textRendering: 'auto',
+        textTransform: 'none',
+        width: '',
+        wordBreak: '',
+      },
+    });
+  });
+});
+
+describe('getHeight', () => {
+  it('returns the height of an HTML node', () => {
+    const element = document.createElement('textarea');
+    // need to set the box sizing property because the function returns null if it is not set
+    element.style.setProperty('box-sizing', 'border-box');
+    // Jest/JSDOM doesn't set scrollHeight, so we need to fake it
+    jest.spyOn(element, 'scrollHeight', 'get').mockImplementation(() => 100);
+    const sizingData = getSizingData(element);
+    // need to force sizingData type to not be null
+    const elementHeight = getHeight(element, sizingData as SizingData);
+    expect(elementHeight).toEqual(102);
+  });
+});
+
+describe('calculateNodeHeight', () => {
+  it('returns the height and rowHeight of an empty textarea', () => {
+    const element = document.createElement('textarea');
+    // need to set the box sizing property because the function returns null if it is not set
+    element.style.setProperty('box-sizing', 'border-box');
+    // Jest/JSDOM doesn't set scrollHeight, so we need to fake it
+    jest.spyOn(element, 'scrollHeight', 'get').mockImplementation(() => 100);
+    const sizingData = getSizingData(element);
+    // need to force sizingData type to not be null
+    const calculatedHeight = calculateNodeHeight(element, sizingData as SizingData, '');
+    expect(calculatedHeight).toEqual([102, 96]);
+  });
+});
+
+describe('useHiddenTextarea', () => {
+  it('creates a visually hidden textarea', () => {
+    render(<SampleComponent />);
+    const hiddenTextarea = screen.getByRole('textbox', {hidden: true});
+    expect(hiddenTextarea).toBeDefined();
```

Comment:
thought(blocker): can you add a test that checks that the hiddenTextarea is removed on unmount please?

```
const { unmount } = render(<Component />);
unmount();
```